src/conn/conn_api.c

```diff
@@ -1822,9 +1821,40 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
               WT_SINGLETHREAD_STRING));
     }
 
+    /*
+     * We own the database home, figure out if we're creating it. There are a few files created when
+     * initializing the database home and we could crash in-between any of them, so there's no
+     * simple test. The last thing we do during initialization is rename a turtle file into place,
+     * and there's never a database home after that point without a turtle file. If the turtle file
+     * doesn't exist, it's a create.
+     */
+    WT_ERR(__wt_turtle_exists(session, &exist));
+    conn->is_new = exist ? 0 : 1;
+
+    /*
+     * If the turtle file exists, the WiredTiger file should exist as well, unless we are salvaging.
+     */
+    WT_ERR(__wt_config_gets(session, cfg, "salvage", &cval));
+    if (is_create && !cval.val && !conn->is_new) {
+        WT_ERR(__wt_fs_exist(session, WT_WIREDTIGER, &exist));
+        if (!exist) {
+            F_SET(conn, WT_CONN_DATA_CORRUPTION);
+            WT_ERR_MSG(session, WT_TRY_SALVAGE, "WiredTiger version file cannot be found");
+        } else {
+            WT_ERR(__wt_open(session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR, 0, &fh));
```

Comment:
Instead of opening, finding file size and then closing. Would it better if we found the size after we create/open the file below? I am thinking we can do the same check without the need to open and closing the file.